include/xtensor/xbuilder.hpp

```diff
@@ -600,61 +633,28 @@ namespace xt
             }
 
             template <class... Args>
-            inline value_type operator()(std::size_t first_idx, Args... args) const
+            inline value_type operator()(Args... args) const
             {
-                return m_source(m_shape_first - first_idx, args...);
+                return access_impl(xindex({ args... }));
             }
 
             template <class It>
             inline value_type element(It first, It last) const
             {
                 // TODO: avoid memory allocation
-                xindex idx(first, last);
-                idx.front() = m_shape_first - idx.front();
-                return m_source.element(idx.begin(), idx.end());
+                return access_impl(xindex(first, last));
```

Comment:
just out of curiosity, I thought the compiler would recognize that it's not necessary to keep the object around and wouldn't copy but instead move the object.